File: AtCoder/ABC/ABC209/C.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int n; cin >> n;
    vector<long long> c(n);
    int mod = 1000000007;
    for (int i = 0; i < n; i++) cin >> c[i];
    sort(c.begin(), c.end());
    int ans = 1;
    for (int i = 0; i < n; i++) ans = ans * max(0LL, c[i] - i) % mod;
    cout << ans << endl;
}

File: AtCoder/ABC/ABC209/C.py
n = int(input())
c = list(map(int,input().split()))
c.sort()
ans = 1
mod = 10 ** 9 + 7
cnt = 0
for i in range(n):
    #print("iのうえ", c[i] - cnt)
    if cnt < c[i]:
        ans *= (c[i] - cnt)
        ans %= mod
        cnt += 1
    else:
        exit(print(0))
    #print(i, ans, cnt) 
print(ans % mod)